### PR TITLE
feat: RandomLinearIllumination and RandomLinearCornerIllumination

### PR DESCRIPTION
**Pull request by vgilabert94**
#### Changes
Add Random illumination augmentation using linear mask.

RandomLinearIllumination:
image

RandomLinearCornerIllumination: 
image

related to https://github.com/kornia/kornia/issues/2721
related to https://github.com/kornia/kornia/issues/941

I'm creating the gradient mask for batching manually, I'm not sure how i can do it without a for loop. If exist an alternative I will change it.

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/augmentation/random_generator/_2d/linear_illumination.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            torch.tensor(-1),
+        ).to(device=_device, dtype=_dtype)
+
+        directions = _adapted_rsampling((batch_size, 1, 1, 1), self.directions_sampler, same_on_batch).to(
+            device=_device,
+            dtype=torch.int8,
+        )
+
+        gradient = torch.zeros(batch_shape)
+        for _b in range(batch_size):
```
Comment (johnnv1) [inline kornia/augmentation/random_generator/_2d/linear_illumination.py:78]: why it's a loop over the batch?

Comment (vgilabert94) [reply kornia/augmentation/random_generator/_2d/linear_illumination.py:78]: No, the batch and direction are independent. I have to create a loop for batch because linspace does not accept batch, is the only way that I thought. And direction in a simple random value between 0 and 4.

Comment (edgarriba) [reply kornia/augmentation/random_generator/_2d/linear_illumination.py:78]: maybe you can workaround this with a meshgrid ?

Comment (vgilabert94) [reply kornia/augmentation/random_generator/_2d/linear_illumination.py:78]: I'm seeing `meshgrid `, and only accept 0D or 1D array as input, so we can't work with channels or batch. I'm not sure how we can do it.

https://discuss.pytorch.org/t/batched-meshgrid/157206
https://stackoverflow.com/questions/72782751/how-to-perform-torch-meshgrid-over-multiple-tensors-in-parallel

Comment (johnnv1) [reply kornia/augmentation/random_generator/_2d/linear_illumination.py:78]: may this do the trick? https://github.com/kornia/kornia/blob/886b9328ae4a1fff39acd953392de80dbc0fd468/kornia/utils/grid.py#L9

Comment (edgarriba) [reply kornia/augmentation/random_generator/_2d/linear_illumination.py:78]: i think so -- side comment, can we make this performance improvement in a follow up PR ? @vgilabert94 we plan to release early next week and would be great to have this patch in.

**Review by johnnv1 — APPROVED** [review]
